### ISSUE TITLE
`api.url` isn't working on aws 

### ISSUE THREAD

**Issue by tsuf239** [winglang/wing#3342]
### I tried this:

```
bring cloud;
let api = new cloud.Api();

api.get("/path", inflight (req: cloud.ApiRequest): cloud.ApiResponse => {

  return cloud.ApiResponse {
    status: 200,
    body: "ok"
  };
});

log(api.url); // logs the token

let apiUrl = api.url;

test "main" {
  log(apiUrl); // a string with " at the beginning and " at the end :(
  log(api.url); // this will be undefined (therefore throws "Unexpected token u in JSON at position 0") see https://github.com/winglang/wing/blob/7bbc81ff26a97b799dafa2a976dab4d32135cd2e/libs/wingsdk/src/shared-tf/tokens.ts#L22-L26
}
```

### This happened:

either getting an error or a URL wrapped in quotation marks 

### I expected this:

getting no errors and valid string 

### Component

Compiler, SDK

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by skorfmann** [comment]
Seems like this is related to https://github.com/winglang/wing/issues/3289

**Comment by tsuf239** [comment]
fixed by #3772
